core/src/executor/select/mod.rs

```diff
@@ -166,40 +186,50 @@ pub async fn select_with_labels<'a>(
     };
 
     let TableWithJoins { relation, joins } = &table_with_joins;
-    let table = Table::new(relation)?;
-
-    let columns = fetch_columns(storage, table.get_name()).await?;
-    let join_columns = stream::iter(joins.iter())
-        .map(Ok::<_, Error>)
-        .and_then(|join| {
-            let table = Table::new(&join.relation);
-
-            async move {
-                let table = table?;
-                let table_alias = table.get_alias();
-                let table_name = table.get_name();
-
-                let columns = fetch_columns(storage, table_name).await?;
+    let (rows, columns) = match relation {
+        TableFactor::Table { .. } => {
+            let columns = fetch_columns(storage, get_name(relation)?).await?;
+            let columns = Rc::from(columns);
+            (
+                Rows::Table(fetch_blended(storage, relation, Rc::clone(&columns)).await?),
+                columns,
+            )
+        }
+        TableFactor::Derived { subquery, alias } => {
+            let (labels, inline_view) = select_with_labels(storage, subquery, None, true).await?;
+            let inline_view = inline_view.try_collect::<Vec<_>>().await?;
+            let labels_rc = Rc::from(labels.to_owned());
+            let labels = Rc::clone(&labels_rc);
+            let rows = inline_view.into_iter().map(move |row| {
+                Ok(BlendContext::new(
+                    &alias.name,
+                    Rc::clone(&labels_rc),
+                    Some(row),
+                    None,
+                ))
+            });
+            (Rows::Derived(stream::iter(rows)), Rc::clone(&labels))
```

Comment:
Here we don't need to collect stream into `Vec`.
```rust
let inline_view = inline_view.try_collect::<Vec<_>>().await?;
```

Rather than that, this can be handled using chaining without evaluation.

```rust
let (labels, rows) = select_with_labels(storage, subquery, None, true).await?;
let labels = Rc::from(labels.clone());
let rows = {
    let labels = Rc::clone(&labels);

    rows.and_then(move |row| {
        let labels = Rc::clone(&labels);
        let alias = &alias.name;

        async move { Ok(BlendContext::new(alias, labels, Some(row), None)) }
    })
};

(Rows::Derived(rows), labels)
```

Reply:
After implementing what you recommanded `fetch_relation_rows`, columns and rows are splited.
Which one is better between above and `fetch_relation_rows` way?